config/main.py

```diff
@@ -2517,6 +2553,244 @@ def remove(ctx, interface_name, ip_addr):
+            ctx.fail("Profile {} doesn't exist or isn't a lossless profile".format(override_profile))
+        profile_full_name = "[BUFFER_PROFILE|{}]".format(override_profile)
+        config_db.set_entry("BUFFER_PG", (interface_name, pg_map), {"profile": profile_full_name})
+    else:
+        config_db.set_entry("BUFFER_PG", (interface_name, pg_map), {"profile": "NULL"})
+    adjust_pfc_enable(ctx, interface_name, pg_map, True)
+
+
+def remove_pg_on_port(ctx, interface_name, pg_map):
+    config_db = ctx.obj["config_db"]
+
+    # Check whether port is legal
+    ports = config_db.get_entry("PORT", interface_name)
+    if not ports:
+        ctx.fail("Port {} doesn't exist".format(interface_name))
+
+    # Remvoe all dynamic lossless PGs on the port
+    existing_pgs = config_db.get_table("BUFFER_PG")
+    removed = False
+    for k, v in existing_pgs.items():
+        port, existing_pg = k
+        if port == interface_name and (not pg_map or pg_map == existing_pg):
+            need_to_remove = False
+            referenced_profile = v.get('profile')
+            if referenced_profile and referenced_profile == '[BUFFER_PROFILE|ingress_lossy_profile]':
+                if pg_map:
+                    ctx.fail("Lossy PG {} can't be removed".format(pg_map))
+                else:
+                    continue
+            config_db.set_entry("BUFFER_PG", (interface_name, existing_pg), None)
+            adjust_pfc_enable(ctx, interface_name, pg_map, False)
+            removed = True
+    if not removed:
+        if pg_map:
+            ctx.fail("No specified PG {} found on port {}".format(pg_map, interface_name))
+        else:
+            ctx.fail("No lossless PG found on port {}".format(interface_name))
+
+
+def adjust_pfc_enable(ctx, interface_name, pg_map, add):
+    config_db = ctx.obj["config_db"]
+
+    # Fetch the original pfc_enable
+    qosmap = config_db.get_entry("PORT_QOS_MAP", interface_name)
+    pfc_enable = qosmap.get("pfc_enable")
+
+    pfc_set = set()
+    if pfc_enable:
+        for priority in pfc_enable.split(","):
+            pfc_set.add(int(priority))
+
+    if pg_map:
+        lower_bound = int(pg_map[0])
+        upper_bound = int(pg_map[-1])
+
+        for priority in range(lower_bound, upper_bound + 1):
+            if add:
+                pfc_set.add(priority)
+            elif priority in pfc_set:
+                pfc_set.remove(priority)
+
+        empty_set = set()
+        pfc_enable = ""
+        if not pfc_set.issubset(empty_set):
+            for priority in pfc_set:
+                pfc_enable += str(priority) + ","
+    elif not add:
+        # Remove all
+        pfc_enable = ""
+    else:
+        ctx.fail("Try to add empty priorities")
+
+    qosmap["pfc_enable"] = pfc_enable[:-1]
+    config_db.set_entry("PORT_QOS_MAP", interface_name, qosmap)
+
+
+#
+# 'buffer' subgroup ('config interface buffer ...')
+#
+@interface.group(cls=clicommon.AbbreviationGroup)
```

Comment:
unit testing missing for interface group. Please add

Reply:
Hi @neethajohn ,
this needs an infrastructure level update.
For unit testing to work for a command, the fixture `@clicommon.pass_db` is required. However, the db objects are generated in interface subcommand instead of passed by `@clicommon.pass_db`.
So we are unable to add unit testing here until it is updated. There isn't any unit test for any of `config interface` commands now for the same cause.
If we would like to enable unit testing for all `config interface` commands, I suggest updating the interface command first and the updating unit test cases. It's better to do it in another PR.
How do you think?

All other comments have been fixed.

Reply:
@lguohan, how do we want to proceed on this? I don't see any unit tests for any of the 'config interface' commands currently

Reply:
Hi @neethajohn @lguohan ,
Can we merge it and add unit test cases for `config interface buffer` when the infrastructure is ready?
Thanks.
Stephen

Reply:
[issue #1301](https://github.com/Azure/sonic-utilities/issues/1301) opened for tracking this.